packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -78,10 +78,12 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
     try {
       return new cxapi.CloudAssembly(appDir);
     } catch (error) {
-      if (error.message.includes(cxschema.VERSION_MISMATCH)) {
+      if (error.versionMismatch && error.manifestVersion) {
```

Comment:
It's probably okay (i.e. I can't think why this will break), but I would advise against changing how this error is identified, and keep it as-is. There is no need to touch this right now, so let's try to avoid unexpected breakage.

Reply:
👍